firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteOverlayDataMigrationManagerTest.java

```diff
@@ -0,0 +1,182 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.local;
+
+import static com.google.firebase.firestore.testutil.TestUtil.deleteMutation;
+import static com.google.firebase.firestore.testutil.TestUtil.deletedDoc;
+import static com.google.firebase.firestore.testutil.TestUtil.doc;
+import static com.google.firebase.firestore.testutil.TestUtil.key;
+import static com.google.firebase.firestore.testutil.TestUtil.map;
+import static com.google.firebase.firestore.testutil.TestUtil.mergeMutation;
+import static com.google.firebase.firestore.testutil.TestUtil.patchMutation;
+import static com.google.firebase.firestore.testutil.TestUtil.setMutation;
+import static java.util.Arrays.asList;
+import static org.junit.Assert.assertEquals;
+
+import com.google.firebase.firestore.FieldValue;
+import com.google.firebase.firestore.auth.User;
+import com.google.firebase.firestore.model.Document;
+import com.google.firebase.firestore.model.MutableDocument;
+import com.google.firebase.firestore.model.mutation.Mutation;
+import java.util.Collections;
+import java.util.List;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE)
+public class SQLiteOverlayDataMigrationManagerTest {
```

Comment:
FYI If you create an "OverlayMigrationRunnable" then you should rename these test to reflect the new name. These tests mostly test the overlay migration and not the manager (but they do test the migration very well!).

Reply:
Keeping it until we resolve the other comment.